### DISCUSSION TITLE
Expected Error Code for TX Failure Due to Coexistence

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#10989] [Q&A] [answered]
Hi,

In scenarios where coexistence mechanisms are in place (e.g., Wi-Fi/BLE coex), and a transmission (TX) attempt fails because another activity is ongoing on the air, what is the expected error code should be returned to OpenThread stack from radio driver?

Specifically, should it be OT_ERROR_CHANNEL_ACCESS_FAILURE, OT_ERROR_ABORT, or any other error?

Thank you for your guidance!

**Comment by abtink** [comment, marked as answer]
The `OT_ERROR_ABORT` is the suggested error in such a case.

Note that the list of accepted errors is documented in `platform/radio.h` for [`otPlatRadioTxDone()`](https://github.com/openthread/openthread/blob/f0d6007cc3a2b85c9c311756da2da9e01ad25a98/include/openthread/platform/radio.h#L921-L936). Other errors MUST not be returned and there are strict checks (`asserts`) within OT core modules for this, e.g., in [`SubMac::HandleTransmitDone()`](https://github.com/openthread/openthread/blob/f0d6007cc3a2b85c9c311756da2da9e01ad25a98/src/core/mac/sub_mac.cpp#L531-L568).

**Reply by testter2** [reply]
Thank you for your quick response
